enhancements/security/security-profiles-operator.md

```diff
@@ -0,0 +1,636 @@
+  Groups:                                       <none>
+Settings:                                       
+  Allow Privileged:                             false
+  Allow Privilege Escalation:                   false
+  Default Add Capabilities:                     <none>
+  Required Drop Capabilities:                   <none>
+  Allowed Capabilities:                         *
+  Allowed Seccomp Profiles:                     *
+  Allowed Volume Types:                         *
+  Allowed Flexvolumes:                          <all>
+  Allowed Unsafe Sysctls:                       *
+  Forbidden Sysctls:                            <none>
+  Allow Host Network:                           true
+  Allow Host Ports:                             true
+  Allow Host PID:                               true
+  Allow Host IPC:                               true
+  Read Only Root Filesystem:                    false
+  Run As User Strategy: RunAsAny                
+    UID:                                        <none>
+    UID Range Min:                              <none>
+    UID Range Max:                              <none>
+  SELinux Context Strategy: RunAsAny  (1)          
+    User:                                       <none>
+    Role:                                       <none>
+    Type:                                       <none>
+    Level:                                      <none>
+  FSGroup Strategy: RunAsAny                    
+    Ranges:                                     <none>
+  Supplemental Groups Strategy: RunAsAny        
+    Ranges:                                     <none>
+
+```
+
+(1) See the Open Questions section
+
+With this profile, the intent is to disallow privileged containers,
+while enabling the usage of custom Seccomp and SELinux profiles.
+
+There needs to be extra validation in place in such a way that
+using the `measured` SCC will also result in workloads
+only being able to use profiles that are installed in the same
+namespace as the workload.
+
+This will prevent workloads trying to:
+
+* Wrongly use profiles that are assumed to be "pre-installed" in the system.
+
+* Share a profile accross namespaces which might lead to errors.
+
+* Maliciously gain privileges by using a profile installed from another
+  namespace.
+
+### OLM integration
+
+As part of OpenShift promoting security as part of the platform, application
+developers need to be able to install Security Profiles when installing operators.
+
+For this, a `securityProfiles` section could be added to the CSV definition as part
+of the `install.spec` section. Following a similar pattern as permissions, it could
+look as follows:
+
+```yaml
+...
+  install:
+    spec:
+      clusterPermissions:
+        - rules:
+          - apiGroups:
+            - ""
+            resources:
+            - nodes
+            verbs:
+            - get
+            - list
+            - watch
+          serviceAccountName: some-sa
+      securityProfiles:
```

Comment:
OLM is intentionally trying to move away from the cluster service version and eventually in a direction where arbitrary kubernetes manifests are dropped into an operator bundle that OLM applies as the installation step. We already have support for a number of other manifests. Is there a reason why we couldn't simply add support in OLM for these manifest types?

Reply:
I was not aware of the OLM moving away from CSVs. Is there an enhancement or doc about this? I can change this proposal to reflect that.

Reply:
this is very out of date, but it does go into some of the high level concepts: https://github.com/operator-framework/enhancements/blob/master/enhancements/csvless-bundles.md

But more specifically, OLM already supports specific resource types other than CSV and CRD in operator bundles today. For example, we support certain Prometheus and ServiceMonitor objects as raw kube manifests applied by the bundle, along with some rbac related resources. I think that, while it might take us a little while before a set of arbitrary kube manifests replace the CSV entirely, we don't want to continue expanding the CSV api and instead OLM should just add these object types as support bundle objects to move this forward.

Reply:
I think there was an EP at some point for something that we call CSV-less bundles. But independent of that we already support a bunch of extra manifests that are applied during install. So we can definitely talk about this at any point.

Reply:
That sounds good to me! I'll remove the reference to the CSV from this EP. Can someone pass me some docs to read that offer examples of other manifests that the OLM supports? That way I could use those as examples for this proposal.

Reply:
@JAORMX see here https://docs.openshift.com/container-platform/4.7/operators/understanding/olm-packaging-format.html#olm-bundle-format_olm-packaging-format

Reply:
@dmesser @kevinrizza the doc makes sense! So, how trivial is to add new types to the supported types? I guess we'd then need to add `SelinuxProfiles` and `SeccompProfiles` as supported types, so folks could then add them to their bundles and the OLM could then install them.

Are there requirements for these types? e.g. required conditions that the objects should have or a status used to interact with the OLM?